odoo/service/security.py

```diff
@@ -17,6 +19,8 @@ def compute_session_token(session, env):
 def check_session(session, env, request=None):
     self = env['res.users'].browse(session.uid)
     expected = self._compute_session_token(session.sid)
+    if 'deletion_time' in session and session['deletion_time'] <= time.time():
```

Comment:
Maybe this should be moved abofe everything, as there is no need to compute the session token if this condition is met. `expected` is unused when `'deletion_time' in session and session['deletion_time'] <= time.time()` so we compute it for nothing